### PR TITLE
fix: Update compute mocks

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

TF uses compute/beta/ endpoint, to avoid log diffs after migrating to Scifi, I made below changes:

The urls should be v1 `https://compute.googleapis.com/compute/v1/`
The self links should have format `https://www.googleapis.com/compute/v1/...`

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by justinsb** [comment]
Could you start the direct controller using the same API version as the existing controllers?

**Comment by gemmahou** [comment]
> Could you start the direct controller using the same API version as the existing controllers?

Per this [comment](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/mockgcp/mockcompute/service.go#L198) I guess we want to switch to v1 eventually? 
Are you suggesting using beta version as the existing controller for easier testing and ensure backward compatibility? 

Other than the API version, the selflink format in compute mocks are inconsistent(should be `https://www.googleapis.com/compute/v1`, b/131629538), I can address that in this PR.

**Comment by justinsb** [comment]
> Are you suggesting using beta version as the existing controller for easier testing and ensure backward compatibility?

Right - I think that should be easier.  Otherwise I think all our tests will start failing with the TF-based controllers?

File: tests/e2e/unified_test.go [hunk 1]
```diff
@@ -475,11 +475,20 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
...
 						for k, v := range pathIDs {
-							url = strings.ReplaceAll(url, "/"+k, "/"+v)
+							u = strings.ReplaceAll(u, "/"+k, "/"+v)
+						}
+
+						// Specific to Compute
+						// Terraform uses the /beta/ endpoints, but mocks and direct controller should use /v1/
+						// This special handling to avoid diffs in http logs.
+						// This can be removed once all Compute resources are migrated to direct controller.
+						basePath := "https://compute.googleapis.com/compute"
```
Comment (justinsb) [inline tests/e2e/unified_test.go:487]: Oh I see!  I guess this is a nice trick also.  @yuwenma  what do you think?

Comment (yuwenma) [reply tests/e2e/unified_test.go:487]: works for me.

**Pull request merged**